src/core/js/models/itemsModel.js

```diff
@@ -0,0 +1,126 @@
+define([
+    'core/js/adapt',
+    'core/js/models/componentModel'
+], function(Adapt, ComponentModel) {
+
+    var ItemsModel = ComponentModel.extend({
+
+        reset: function(type, force) {
+            this.resetItems();
+            ComponentModel.prototype.reset.call(this, type, force);
+        },
+
+        resetItems: function() {
+            _.each(this.get('_items'), function(item) {
+                item._isVisited = false;
+                item._isActive = false;
+            });
+        },
+        
+        getItemCount: function() {
+            return this.get('_items').length;
+        },
+
+        getItemAtIndex: function(index) {
+            return this.get('_items')[index];
+        },
+
+        getVisitedItems: function() {
+            return _.filter(this.get('_items'), function(item) {
+                return item._isVisited;
+            });
+        },
+
+        setItemAtIndexAsVisited: function(index) {
+            var item = this.get('_items')[index];
+            if (item) {
+                item._isVisited = true;
+            }
+        },
+
+        areAllIItemsCompleted: function() {
+            return (this.getVisitedItems().length == this.getItemCount());
+        },
+
+        checkCompletionStatus: function() {
+            if (this.areAllIItemsCompleted()) {
+                this.setCompletionStatus();
+            }
+        },
+
+        resetActiveItems: function(trigger) {
+            var items = this.get('_items');
+
+            _.each(items, function(item) {
+                item._isActive = false;
+            });
+
+            if (trigger !== false) {
+                this.trigger('change:_items:_isActive', this, items);
+            }
+        },
+
+        getActiveItems: function() {
+            return _.filter(this.get('_items'), function(item) {
+                return item._isActive;
+            });
+        },
+        
+        getFirstActiveItem: function() {
+            return this.getActiveItems()[0];
+        },
+        
+        getActiveItemsCount: function() {
+            return this.getActiveItems().length;
+        },
+        
+        getActiveItemsIndexes: function() {
+            var indexes = [];
+            var items = this.get('_items');
+            
+            for (var i = 0, l = items.length; i < l; i++) {
+                if (items[i]._isActive) {
+                    indexes.push(i); 
+                }
+            }
+
+            return indexes;
+        },
+
+        getFirstActiveItemIndex: function() {
+            return this.getActiveItemsIndexes()[0];
+        },
+
+        setItemAtIndexAsActive: function(index, trigger) {
```

Comment:
the shorter ones would be my preference, i was using this today and it seemed a bit weird having ``getItemIndex`` or ``setItemAtIndexAsActive`` rather than ``setItemActive``, which is nice and short.